feat: separate stubs into their own wheel (#154)
algorandfoundation/puya: stubs/README.md

> # AlgoPy - API for writing Algorand Python smart contracts

Reviewer: ```suggestion
# Algorand Python framework - API for writing Algorand Python smart contracts
```

Reviewer: AlgoPy isn't the name of it, we don't want to have confusion by introducing extra terminology, `algopy` is simply the abbreviated import path for Algorand Python.

Reviewer: unsure if we call it "framework" or "APIs"?

Reviewer 2: It's both, but framework is the broader term so that gets my +1

## Installation

### API
The minimum supported Python version for writing Algorand Python smart contracts is 3.12.

> You can install AlgoPy from PyPI into your project virtualenv:

Reviewer: ```suggestion
You can install the Algorand Python framework from PyPI into your project virtualenv:
```

```shell

> pip install algopy-stubs

Reviewer: ```suggestion
pip install algorand-python
```

```
If you're using poetry for dependency and virutalenv management, you can add it that way with

> `poetry add algopy-stubs`.

Reviewer: ```suggestion
poetry add algorand-python
```

> 

Reviewer: ```suggestion
Once you have installed the Algorand Python framework, you can access the type definitions from the `algopy` module, e.g.

\`\`\`python
from algopy import Contract
\`\`\`
```

### Compiler

To compile a smart contract, you will also need the PuyaPy compiler, this can either be installed globally (e.g. `pipx install puya`), or
as a dev dependency in your project virtualenv (e.g. `poetry add puya --group=dev`)

## Usage

To compile a contract or contracts, just supply the path(s) - either to the .py files themselves,
or the containing directories. In the case of containing directories, any contracts discovered
therein will be compiled, allowing you to compile multiple contracts at once. You can also supply
more than one path at a time to the compiler.

> e.g. `puyapy my_project/` or `puyapy my_project/contract.py` will work to compile a single contract.

Reviewer: This shouldn't be here, it's covered in the other documentation. Maybe we can have a Usage one liner that links back to the language guide and/or main readme?
